### ISSUE TITLE
[2.8] [BUG] vsphere 8 using rke2/k3s - if vsphere DRS maxVcpusPerCore limit is reached on host, rancher does not clean up vsphere VMs 

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#46487]
backport of https://github.com/rancher/rancher/issues/44212 in the v2.8.x release line

**Comment by snasovich** [comment]
This will be done indirectly as part of bumping `rancher/machine` for https://github.com/rancher/rancher/issues/46242

**Comment by jiaqiluo** [comment]
The fix can be validated on the latest v2.8-head image

**Comment by slickwarren** [comment]
tested on rancher on v2.8-head: 
```
 Invalid virtual machine configuration. A virtual IOMMU should always be enabled for a VCPU count of more than 128. 
```

```
 The total number of virtual CPUs present or requested in virtual machines' configuration has exceeded the limit on the host: 192. 
```

I was able to do this by:
giving a rancher downstream cluster an extremely large cpu count
using a reasonable cpu count for nodes, but having more nodes * cpu than the vsphere setup is configured for. 


In all of these cases, the machines are now cleaned up on rancher and on vsphere 8.
